resources/todomvc/big-dom-generator/src/tree-generator.js

```diff
@@ -0,0 +1,94 @@
+import { LCG } from "random-seedable";
+import { DEFAULT_SEED_FOR_RANDOM_NUMBER_GENERATOR, MAX_GENERATED_DOM_DEPTH, MAX_NUMBER_OF_CHILDREN, PROBABILITY_OF_HAVING_CHILDREN, TARGET_SIZE, MIN_NUMBER_OF_MAX_DEPTH_BRANCHES } from "./../params";
+
+const random = new LCG(DEFAULT_SEED_FOR_RANDOM_NUMBER_GENERATOR);
+
+/**
+ * Generates the blueprint of the tree-view side panel for the complex DOM shell with expandable and non-expandable items.
+ * It starts with the minimum number of maximum-depth branches and randomly adds
+ * children to the nodes in a breadth-first manner.
+ * The weight parameters represent how many DOM nodes are generated for each type of node.
+ * @param {Object} options - The options object.
+ * @param {number} options.expandableItemWeight - The weight for the "expandableItem" node type. <li></li> with ChevronRight svg.
+ * @param {number} options.nonExpandableItemWeight - The weight for the "nonExpandableItem" node type. <li></li> TaskListIcon svg.
+ * @returns {Object} The generated tree structure with two possible values for the "type" property: "expandableItem" or "nonExpandableItem"
+ * Example structure:
+ * {
+ *    type: "expandableItem",
+ *    children: [
+ *      {
+ *         type: "expandableItem",
+ *         children: [
+ *           {
+ *             type: "nonExpandableItem",
+ *             children: []
+ *           }
+ *         ]
+ *      }
+ *    ]
+ * }
+ **/
+export const generateTreeHead = ({ expandableItemWeight, nonExpandableItemWeight }) => {
+    const treeHead = { type: "expandableItem", children: [] };
+    const nodeWeight = { expandableItem: expandableItemWeight, nonExpandableItem: nonExpandableItemWeight };
+
+    let totalNodes = expandableItemWeight;
+    for (let i = 0; i < MIN_NUMBER_OF_MAX_DEPTH_BRANCHES; i++) {
+        let currentDepth = 0;
+        let currentNode = treeHead;
+        while (currentDepth < MAX_GENERATED_DOM_DEPTH) {
+            let childType = "expandableItem";
+            currentNode.children.push({ type: childType, children: [] });
+            currentNode = currentNode.children[currentNode.children.length - 1];
+            currentDepth++;
+            totalNodes += nodeWeight[childType];
+        }
+        // Ensure the last node is not an expandableItem.
```

Comment:
I'm not sure this is correct: if expandableItem can have 0 children, then it can also be the last node. So my suggestion would be to remove these few lines: they bring nothing to the workload but confusion. (optional)

Reply:
Removed.